File: C#/WindowsPhone/LGame-XNA-lib/Loon.Action.Avg.Drama/Expression.cs
namespace Loon.Action.Avg.Drama
{
    using System;
    using Loon.Core;

    public abstract class Expression
    {

        // 默认变量1,用于记录当前选择项
        public const string V_SELECT_KEY = "SELECT";

        // 左括号
        public const string BRACKET_LEFT_TAG = "(";

        // 右括号
        public const string BRACKET_RIGHT_TAG = ")";

        // 代码段开始标记
        public const string BEGIN_TAG = "begin";

        // 代码段结束标记
        public const string END_TAG = "end";

        // 代码段调用标记
        public const string CALL_TAG = "call";

        // 缓存刷新标记
        public const string RESET_CACHE_TAG = "Reset";

        // 累计输入数据标记
        public const string IN_TAG = "in";

        // 累计输入数据停止（输出）标记
        public const string OUT_TAG = "out";

        // 多选标记
        public const string SELECTS_TAG = "selects";

        // 打印标记
        public const string PRINT_TAG = "print";

        // 随机数标记
        public const string RAND_TAG = "rand";

        // 设定环境变量标记
        public const string SET_TAG = "set";

        // 载入内部脚本标记
        public const string INCLUDE_TAG = "include";

        // 条件判定标记
        public const string IF_TAG = "if";

        // 条件判定结束标记
        public const string IF_END_TAG = "endif";

        // 转折标记
        public const string ELSE_TAG = "else";

        // 以下为注视符号
        public const string FLAG_L_TAG = "//";

        public const string FLAG_C_TAG = "#";

        public const string FLAG_I_TAG = "'";

        public const string FLAG_LS_B_TAG = "/*";

        public const string FLAG_LS_E_TAG = "*/";

        public const string FLAG = "@";

        public const string FLAG_SAVE_TAG = "save";

        public const string FLAG_LOAD_TAG = "load";

        public const char FLAG_CHAR = '@';

    }
}

File: Java/old/AWT_ver/src/org/loon/framework/javase/game/action/avg/command/Expression.java
package org.loon.framework.javase.game.action.avg.command;

import java.util.Random;

import org.loon.framework.javase.game.core.LSystem;

/**
 * Copyright 2008 - 2009
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 * 
 * http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 * 
 * @project loonframework
 * @author chenpeng
 * @email：ceponline@yahoo.com.cn
 * @version 0.1
 */
public interface Expression {

	// 全局随机数
	Random GLOBAL_RAND = LSystem.random;

	// 默认变量1,用于记录当前选择项
	String V_SELECT_KEY = "SELECT";

	// 左括号
	String BRACKET_LEFT_TAG = "(";

	// 右括号
	String BRACKET_RIGHT_TAG = ")";

	// 代码段开始标记
	String BEGIN_TAG = "begin";

	// 代码段结束标记
	String END_TAG = "end";

	// 代码段调用标记
	String CALL_TAG = "call";

	// 缓存刷新标记
	String RESET_CACHE_TAG = "reset";
	
	// 累计输入数据标记
	String IN_TAG = "in";

	// 累计输入数据停止（输出）标记
	String OUT_TAG = "out";

	// 多选标记
	String SELECTS_TAG = "selects";

	// 打印标记
	String PRINT_TAG = "print";

	// 随机数标记
	String RAND_TAG = "rand";

	// 设定环境变量标记
	String SET_TAG = "set";

	// 载入内部脚本标记
	String INCLUDE_TAG = "include";

	// 条件判定标记
	String IF_TAG = "if";

	// 条件判定结束标记
	String IF_END_TAG = "endif";

	// 转折标记
	String ELSE_TAG = "else";

	// 以下为注视符号
	String FLAG_L_TAG = "//";

	String FLAG_C_TAG = "#";

	String FLAG_I_TAG = "'";

	String FLAG_LS_B_TAG = "/*";

	String FLAG_LS_E_TAG = "*/";

	String FLAG = "@";

	char FLAG_CHAR = FLAG.toCharArray()[0];

}
